packages/gitbook/src/lib/pages.ts

```diff
@@ -1,53 +1,99 @@
+import type { GitBookLinker } from '@/lib/links';
+import { joinPath } from '@/lib/paths';
+import type { FlatPageEntry } from '@/lib/sitemap';
 import {
     type Revision,
     type RevisionPage,
     type RevisionPageDocument,
     type RevisionPageGroup,
     RevisionPageType,
 } from '@gitbook/api';
+import type { ListItem, Paragraph, RootContent } from 'mdast';
 
 export type AncestorRevisionPage = RevisionPageDocument | RevisionPageGroup;
 
+interface ResolvePageOptions {
+    /**
+     * Whether to include page groups in the results.
+     */
+    includePageGroup?: boolean;
+}
+
 /**
- * Resolve a page path to a page document.
+ * Resolves a page path to find the corresponding page document or group.
  */
 export function resolvePagePath(
     rootPages: Revision['pages'],
-    pagePath: string
-): { page: RevisionPageDocument; ancestors: AncestorRevisionPage[] } | undefined {
-    const iteratePages = (
-        pages: RevisionPage[],
-        ancestors: AncestorRevisionPage[]
-    ): { page: RevisionPageDocument; ancestors: AncestorRevisionPage[] } | undefined => {
-        for (const page of pages) {
-            if (page.type === RevisionPageType.Link || page.type === RevisionPageType.Computed) {
-                continue;
-            }
+    pagePath: string,
+    options?: { includePageGroup: true }
+):
+    | { page: RevisionPageGroup | RevisionPageDocument; ancestors: AncestorRevisionPage[] }
+    | undefined;
+export function resolvePagePath(
```

Comment:
I'm usually not a big fan of functions with multiple signatures.

I don't know if possible in this case, but it's usually better to have composition:

```
function resolvePagePath(rootPages, pagePath) {
   const page = resolveAnyPagePath(rootPages, pagepath)
   if (page.type === 'group') {
     return resolveFirstPageDocument(...);
   }

   return page;
}

function resolveAnyPagePath(...)
```